docs/compile.md

```diff
@@ -188,7 +188,69 @@ It's up to you to decide what the output is.
+
+A library that defines resources as classes using the Base Object class (BaseObj) is required. These can then be utilized within components to render output. 
+
+The following functions are provided by the class BaseObj.
+
+Method definitions:<br/>
+
+* new: Provides parameter checking capabilities<br/>
+* body: Enables in-depth parameter configuration<br/>
+
+Method functions:<br/>
+
+* root: Defines values that will be compiled into the output<br/>
+* need: Ability to check & define input parameters<br/>
+* update_root: Updates the template file associated with the class<br/>
+
+A class can be a resource such as a kubernetes Deployment as shown here:
+
+```
+class Deployment(BaseObj):
+    def new(self):
+        self.need("name", "name string needed")
+        self.need("labels", "labels dict needed")
+        self.need("containers", "containers dict needed")
+        self.update_root("lib/kubelib/deployment.yml")
+
+    def body(self):
+        self.root.metadata.name = self.kwargs.name
+        self.root.metadata.namespace = inv.parameters.target_name
+        self.root.spec.template.metadata.labels = self.kwargs.labels
+        self.root.spec.template.spec.containers = self.kwargs.containers
+```
+
+The deployment is an BaseObj which has two main funcations. New can be used to perform parameter validation & template compilation. Body is utilized to set those parameters to be rendered. `self.root.metadata.name` is a direct reference to a key in the corresponding yaml.
+
+We have established that you may define a library which holds information on classes that represent resource objects. The library is then utilized by defined components to generate the required output.
+
+Here we import `kubelib` using `load_from_search_paths`. We then use kubelib to access the defined object classes. In this instance the Deployment & Service resource class.
+
+```
+...
+kubelib = kadet.load_from_search_paths("kubelib")
+...
+name = "nginx"
+labels = kadet.BaseObj.from_dict({"app": name})
+nginx_container = kubelib.Container(
+    name=name, image=inv.parameters.nginx.image, ports=[{"containerPort": 80}]
+)
+...
+def main():
+    output = kadet.BaseObj()
+    output.root.nginx_deployment = kubelib.Deployment(name=name, labels=labels, containers=[nginx_container])
+    output.root.nginx_service = kubelib.Service(
+        name=name, labels=labels, ports=[svc_port], selector=svc_selector
+    )
+    return output
+```
+
+Kadet uses a library called addict to organise the parameters inline with the yaml templates.
+As shown above we create a BaseObject named output. We update the root of this output with the data structure returned from kubelib. This output is what is then returned to kapitan to be compiled into the desired output type.
+
+For a deeper understanding of this input type please review the proposal document at [kadet](/kap_proposals/kap_0_kadet/) & the examples located at `examples/kubernetes/components/nginx`.
```

Comment:
Are you sure the link for `/kap_proposals/kap_0_kadet/` will be rendered and work? It might have to start with `./` ?

Reply:
I believe it will. When I prepend `./` it attempts to access `/compile/kap_proposals/kap_0_kadet/`